File: nodejs/kyc_pf.js
// FonteData - KYC Pessoa Fisica em Node.js
// Custo estimado: ~R$ 9,00 por pessoa
// Docs: https://fontedata.com/docs

const API_KEY = 'fd_live_SUA_CHAVE';
const BASE_URL = 'https://app.fontedata.com/api/v1/consulta';

const CHECKS = {
  identidade:   'receita-federal-pf',
  pep:          'pep-exposicao',
  ceis:         'ceis-sancoes',
  ofac:         'ofac-sancoes',
  antecedentes: 'antecedentes-criminais',
  processos:    'processos-agrupada',
  mandados:     'cnj-mandados-prisao',
};

async function kycPF(cpf) {
  const cpfLimpo = cpf.replace(/[\.\-]/g, '');
  const results = await Promise.all(
    Object.entries(CHECKS).map(async ([name, ep]) => {
      try {
        const res = await fetch(`${BASE_URL}/${ep}/${cpfLimpo}`, { headers: { 'X-API-Key': API_KEY } });
        const status = res.status === 200 ? 'OK' : 'FALHOU';
        console.log(`  [${status}] ${name}: custo=R$${res.headers.get('x-request-cost') || 'N/A'}`);
        return { name, status: res.status, data: res.ok ? await res.json() : null };
      } catch (e) {
        return { name, status: 'error', error: e.message };
      }
    })
  );
  return Object.fromEntries(results.map(r => [r.name, r]));
}

const cpf = process.argv[2] || '12345678900';
console.log(`KYC PF para CPF ${cpf}:\n`);
kycPF(cpf).then(r => console.log(`\n${Object.keys(r).length} checks concluidos.`));

File: python/kyc_pf.py
# FonteData - KYC completo Pessoa Fisica
# Custo estimado: ~R$ 9,00 por pessoa
# pip install requests
import requests, sys
from concurrent.futures import ThreadPoolExecutor, as_completed

API_KEY = "fd_live_SUA_CHAVE"
BASE_URL = "https://app.fontedata.com/api/v1/consulta"
HEADERS  = {"X-API-Key": API_KEY}

CHECKS = {
    "identidade":     "receita-federal-pf",
    "pep":            "pep-exposicao",
    "pep_estendida":  "pep-estendida",
    "ceis":           "ceis-sancoes",
    "ofac":           "ofac-sancoes",
    "onu":            "onu-sancoes",
    "antecedentes":   "antecedentes-criminais",
    "processos":      "processos-agrupada",
    "mandados":       "cnj-mandados-prisao",
    "tse":            "tse-situacao",
}


def _check(name, endpoint, cpf):
    try:
        r = requests.get(f"{BASE_URL}/{endpoint}/{cpf}", headers=HEADERS, timeout=30)
        return name, {"status": r.status_code, "data": r.json() if r.ok else None, "cost": r.headers.get("X-Request-Cost")}
    except Exception as e:
        return name, {"status": "error", "error": str(e)}


def kyc_pf(cpf: str) -> dict:
    cpf = cpf.replace(".", "").replace("-", "")
    results = {}
    with ThreadPoolExecutor(max_workers=5) as ex:
        futures = {ex.submit(_check, n, ep, cpf): n for n, ep in CHECKS.items()}
        for f in as_completed(futures):
            name, result = f.result()
            results[name] = result
            icon = "OK" if result["status"] == 200 else "FALHOU"
            print(f"  [{icon}] {name}: custo=R${result.get('cost','N/A')}")
    return results


if __name__ == "__main__":
    cpf = sys.argv[1] if len(sys.argv) > 1 else "12345678900"
    print(f"KYC PF para CPF {cpf}:\n")
    kyc_pf(cpf)
